request/util.js

```diff
@@ -121,7 +122,7 @@ define([
 			query = options.query;
 
 		if(data && !skipData){
-			if(typeof data === 'object' && !(data instanceof ArrayBuffer || data instanceof Blob )){
+			if(typeof data === 'object' && (!(has('native-xhr2')) || !(data instanceof ArrayBuffer || data instanceof Blob ))){
```

Comment:
We have a feature test for blob ( https://github.com/dojo/dojo/blob/master/request/xhr.js#L34 ) but not for ArrayBuffer as the only implementation that is buggy was either Safari 7 (so we basically implemented Blob support using ArrayBuffer), e.g. https://github.com/dojo/dojo/commit/fb7a76cb0a1dd85463dc419d0814d8c8b9877125 . My goal with this was to reuse one of the existing feature tests given that we have several feature tests defined in https://github.com/dojo/dojo/blob/master/request/xhr.js#L10-L43 .